### DISCUSSION TITLE
Create node/disk tags via CLI?

### DISCUSSION THREAD

**Discussion by pandalec** [longhorn/longhorn#4734] [Q&A] [answered]
Hey!
Is there any option to create node and/or disk tags via cli/kubectl? We have a lot of nodes and want to distribute longhorn storage only where the pod has its affinity to run on. I only see the way via [Longhorn UI](https://longhorn.io/docs/1.3.1/volumes-and-nodes/storage-tags/#overview).

Best regards

**Comment by weizhe0422** [comment, marked as answer]
The tags could be update/add via [kubectl patch](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/). And there two way to specify the value, here are the sample for referenced.
 1. Specified with JSON string: 
      `kubectl patch node.longhorn.io <your_node_name> --type=json -p='[{"op": "replace", "path": "/spec/tags", "value":["WORKER_NODE"]},{"op": "replace", "path": "/spec/disks/<your_disk_name>/tags", "value":["SSD","FAST"]}]' -n longhorn-system`
 2. Specified with yaml file
    1. Create yaml file, e.g. `patch_tags.yaml `
          ```yaml
          spec:
            disks:
              <your_disk_name>: # Specify the disk name
                tags: # Tags for the disk
                - SSD
                - FAST
            tags: # Tags for the node
            - WORKER_NODE
    2. Apply the yaml file: `k patch node.longhorn.io c1-worker1 --patch-file ./patch_tags.yaml -n longhorn-system --type merge`

Hope it can help you to solve it.

**Comment by pandalec** [comment]
Thank you very much! :)
